DataStore2/init.lua

```diff
@@ -182,26 +182,21 @@ end
 function DataStore:SetValidator(validator)
 	assert(
 		type(validator) == "function",
-		"function expected, got " .. type(validator)
+		"function expected, got " .. typeof(validator)
 	)
 	self.validator = validator
 end
 
-local function AssertValidatorWithDefaultError(validator, input, defaultError)
-    local success, err = pcall(validator, input)
-
-    -- if success but result is false
-    if success and err ~= true then
-        error(defaultError)
-    -- if not success and there is an error message
-    elseif not success and err ~= nil then
-        error(err)
-    end
+local function assertValidatorWithDefaultError(validator, input, defaultError)
+	local isValid = validator(input)
+	if not isValid then
+		error(defaultError) 
+	end
```

Comment:
This is essentially the same as you did before.

A validator should be able to return `false, message` and have that message print.